src/content/docs/components/building-components.mdx

```diff
@@ -0,0 +1,159 @@
+---
+title: Building components
+---
+
+import { TabItem, Tabs } from '@astrojs/starlight/components'
+
+Components are built with a builder, which is a JavaScript class, which represents it. Within this builder you can customize the component such it's text or it's color.
+
+There are diffrent types of components' builders with diffrent customization but all have a common property: the custom Id. A unique identifier which is used for handling the interactions of the component.
+
+## ActionRow
+
+All the builders are sent within an [`ActionRow`](/api/classes/actionrow). Each message can contain a maxium of 5 actions row.
+
+Here is how we create a basic ActionRow:
+
+```ts {"Set all the components within this ActionRow": 4-5} {"Add components to the ActionRow": 6-7} showLineNumbers copy
+
+import { ActionRow } from 'seyfert';
+
+const row = new ActionRow()
+
+.setComponents()
+
+.addComponents()
+
+```
+
+## Building each component
+
+Now we are going to build each type of component and set it within an ActionRow:
+
+<Tabs>
+<TabItem label="button.ts">
+
+### Buttons
+
+An ActionRow shouldn't contain more than 5 buttons.
+
+```ts showLineNumbers copy
+
+import {
+    ActionRow,
+    Button
+} from 'seyfert';
+
+import { ButtonStyle } from 'discord-api-types';
+
+const button = new Button()
+.setCustomId('first-button')
+.setStyle(ButtonStyle.Primary)
+.setLabel('First Button')
+
+const row = new ActionRow()
+.setComponents(button)
```

Comment:
```suggestion
const row = new ActionRow<Button>().setComponents(button)
```
It is very common to get a type error here if you do not specify the component that will be in the action row.